### DISCUSSION TITLE
FTP pb_data/storage not working

### DISCUSSION THREAD

**Discussion by HEAVYPOLY** [pockethost/pockethost#236] [Q&A]
Hi, I'm trying to copy files from local to pockethost pb_data/storage via ftp

I can see the pb_data/storage folder on filezilla but when I try to drop files there I get this:

Response: 	150 File status okay; about to open data connection

550 ENOENT: no such file or directory, open '/home/pockethost/data/u06yjvthnoqy6bc/pb_data/storage/69oiwu0s14fw172/rbo1m3awel1iutj/filename_j5wWhkgsUq.png'

I've tried with migration mode turned on and off.  Am I missing something else?

**Comment by benallfree** [comment]
@HEAVYPOLY I think we've seen this one before in #186. I'll need to check into it, this is probably a legitimate bug.

**Comment by HEAVYPOLY** [comment]
Thanks!, loving pb/ph so far!

**Comment by benallfree** [comment]
@HEAVYPOLY I am not able to reproduce your issue. I was able to successfully upload and download to `pb_data/static`.

**Comment by benallfree** [comment]
@HEAVYPOLY I just checked the server file system and the directory `pb_data/storage/69oiwu0s14fw172/rbo1m3awel1iutj` indeed does not exist, but other directories do.

**Comment by benallfree** [comment]
@HEAVYPOLY Oops, update - I can repro by uploading a folder instead of an individual file. Sit tight, I'll fix this :)

**Comment by benallfree** [comment]
@HEAVYPOLY I've updated the FTP code. Now it works with folder uploads as well. I also patched the security problem where it was showing a PocketHost server internal path. Let me know how this works for you!

**Comment by HEAVYPOLY** [comment]
Thanks!  

Uploading and download from storage folder seems to be working, but I can't see any of the files in pb dashboard

<img width="1484" alt="image" src="https://github.com/benallfree/pockethost/assets/24432677/8fed15a5-e7d2-4c38-aba6-8c5d6e912c47">

**Reply by benallfree** [reply]
@HEAVYPOLY I believe PocketBase requires you to create records in the dashboard and associate them with the uploaded content. It will not automatically index your uploaded files and show it in the admin. Is that what you mean?

@ganigeorgiev may be able to explain more about how PocketBase manages storage files.

**Comment by HEAVYPOLY** [comment]
Ah, you're right I mistakenly deleted the records, will try again

**Comment by HEAVYPOLY** [comment]
also tried to download pb_data/data.db and getting 'critical file transfer error' on filezilla
<img width="365" alt="image" src="https://github.com/benallfree/pockethost/assets/24432677/8e6e918a-6bad-4b60-8adf-640411dc01dc">

**Reply by benallfree** [reply]
Thanks for testing this so much. I will need to look into this one deeper. It's probably not a good idea to download `pb_data/*.db` without first putting the instance in Maintenance Mode.

**Reply by benallfree** [reply]
I'm not able to repro the `Critical file transfer error`, so I'm wondering if it's because your PocketBase instance was running and the file was locked? It shouldn't matter, but maybe?

**Comment by HEAVYPOLY** [comment]
Strangely if I try to download pb_data folder instead it works!

Also downloading pb_data, then deleting data on pb dashboard, then uploading pb_data worked to restore everything.  Thanks so much for the quick work!

I'm not sure if it was locked or not, how would I tell?  I didn't change maintenance mode in either case( folder or data.db).
